vector/src/androidTest/java/im/vector/app/ClearCurrentSessionRule.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.test.platform.app.InstrumentationRegistry
+import im.vector.app.features.analytics.store.AnalyticsStore
+import kotlinx.coroutines.runBlocking
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import kotlin.reflect.KClass
+
+class ClearCurrentSessionRule : TestWatcher() {
+    override fun apply(base: Statement, description: Description): Statement {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        runBlocking {
+            reflectAnalyticDatastore(context).edit { it.clear() }
+            runCatching {
+                val holder = (context.applicationContext as VectorApplication).activeSessionHolder
+                holder.getSafeActiveSession()?.signOutService()?.signOut(true)
+                (context.applicationContext as VectorApplication).vectorPreferences.clearPreferences()
```

Comment:
`VectorPreferences.clearPreferences`  doesn't do a full clean up. There are plenty of things which are kept after logout and could affect tests actually. Is it ok?

Reply:
I tried deleting via the `DefaultSharedPrefs` and it actually caused more trouble as it resets other legacy flags like the migration from Riot to Element (which causes a popup to be shown)